### PR TITLE
style: add ESLint to the project

### LINKED ISSUE #155: Add Linter
At present, there is no linter in project, add one. You might want to use [ESLint](https://eslint.org/).

### PR DESCRIPTION
**Pull request by rohitgeddam**
add eslint to the project to enforce good coding style and catch
errors early. Following airbnb style guide.

Fixes: #155

### CODE REVIEW

**Comment by rohitgeddam** [comment]
If you need me to add specific rules in eslint config please let me know.

File: package.json [hunk 1]
```diff
@@ -6,7 +6,8 @@
   "scripts": {
     "start": "node app.js",
     "test:watch": "mocha --watch ./tests/**/*.test.js",
-    "test": "mocha ./tests/**/*.test.js"
+    "test": "mocha ./tests/**/*.test.js",
+    "lint": "eslint"
```
Comment (LalitNM) [inline package.json]: You missed something here.

**Review by LalitNM — COMMENTED** [review]
Let's wait for @sesamechicken's review, meanwhile you can update contributing guidelines.

File: package.json (hunk 1 above)
Comment (sesamechicken) [reply package.json]: @LalitNM What was missed? Code reviews should be as specific as possible 😃

Comment (rohitgeddam) [reply package.json]: @LalitNM can you please point out what I missed

Comment (sesamechicken) [reply package.json]: The only thing I see that you could add would be a dir for eslint to look. Also, I find Airbnb's rules a little restrictive and overly-aggressive in some areas. I don't know who to the official owner of this project is, but linting rules are opinionated and they should chime in.

As for your PR, for the job to run, you want to tell it where to look, so line 10 would look like this:
`"lint": "eslint ./public/**/*.js ./serve/**/*.js"`

Comment (rohitgeddam) [reply package.json]: Should I change style guide from airbnb to standard one.

Comment (LalitNM) [reply package.json]: > Should I change style guide from airbnb to standard one.

Leave it to Airbnb's rules are fine.

Comment (LalitNM) [reply package.json]: > Should I change style guide from airbnb to standard one.

Leave it to Airbnb.

**Comment by LalitNM** [comment]
@rohitgeddam I missed a few changes, listing them here.
1. Add tests folder for linting, use `.js` as extension.
2. Add main `app.js` for linting.
3. Move the content of `eslintrc.json` to `package.json`(Do not add an extra file).

@rohitgeddam just add one line in the [CONTRIBUTING.md](https://github.com/KamandPrompt/baat-cheet/blob/master/CONTRIBUTING.md) in the section 'Submitting a Pull Request' about running `npm run lint`. Before committing the changes and tell the contributor to fix all the errors before commiting.

**Comment by sesamechicken** [comment]
> 3. Move the content of `eslintrc.json` to `package.json`(Do not add an extra file).

Why? This is a great separation of concerns and keeps the `package.json` from growing out of control, because you're going to have unit test settings, as well.

**Comment by LalitNM** [comment]
> This is a great separation of concerns and keeps the package.json from growing out of control

Fair enough.

**Comment by sesamechicken** [comment]
Where is this at?

**Comment by LalitNM** [comment]
> add one line in the CONTRIBUTING.md in the section 'Submitting a Pull Request' about running `npm run lint`.

I thought about this and reached to conclusion that we should do this after applying linting in all files.

**Comment by LalitNM** [comment]
Merging this, will open another issue for applying linting changes in all files.

**Pull request merged**